src/containers/Review/assignment/AssignmentSubmit.tsx

```diff
@@ -51,21 +54,22 @@ const AssignmentSubmit: React.FC<AssignmentSubmitProps> = ({
 
       if (found) {
         found.sectionCodes.push(sectionCode)
-        if (!!unassignment && found?.unassignmentId != unassignment.id)
+        if (!!unassignment && found?.unassignment.id != unassignment.id)
```

Comment:
This statement is checking if there is **more than one** review assignment to be unassigned. For example if you are changing the reviewer of section 1 and 2, but they belong to 2 different review_assignment (different reviewers). This is saying it will fail! So we should probably prevent in the UI to re-assign from **more than one** review. 

One idea is, after you select one **new** reviewer fro a section, we disable other sections that belongs to another review_assigment. TO garantee this is only going to run to unassign the reviewer from **one** review_assignment

Reply:
Okay, I just tried it to see what would happen. Yes, you're correct, it only does one unassignment, but it still assigns both sections to new reviewer.

Surely the most sensible thing to do would be to make it do both unassignments correctly? It would be confusing if the UI didn't let you do something that seems like an obvious thing to do.

I'll take a look now and see how easy it is to fix.

Reply:
It also causes a problem when you swap two reviewers around at the same time. Instead of them being swapped, one ends up with both sections and the other none.

Not sure best way to fix right now. I *don't* think we should change the UI, since it seems like a reasonable thing to be able to expect to do (swap two reviewers around), *and*, I think it would be a lot of work just figuring out the UI logic to restrict this correctly. Seems like it would be better to re-visit the actual assign/unassign logic. 

I have one thought -- what if we don't do any "unassigning" in the "reassign" hook, but just use the existing "Unassign" hook. Then we just run the unassign funcion as many times as we need, then run "re-assign". ? Would that work?

Reply:
> I have one thought -- what if we don't do any "unassigning" in the "reassign" hook, but just use the existing "Unassign" hook. Then we just run the unassign funcion as many times as we need, then run "re-assign". ? Would that work?

This would require a bit of thought to get right. We can discuss next time we chat.